### PR TITLE
Fix HCA-AIQ E2E-spec flaky tests

### PR DESCRIPTION
**Pull request by tlei123**
## Description

Fix Health Care Application (HCA) American Indian Question (AIQ) E2E-spec flakty tests.

## Original issue(s)
[n/a]

## Testing done
 - No functional app-code made.
 - Repeatedly ran updated spec locally, and successfully posted results to TestRail.  [See screenshot below.]

## Screenshots
image

## Acceptance criteria
- [ ] Code-reviewer(s) approved changes.

## Definition of done
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms-system/src/js/components/FormTitle.jsx [hunk 1]
```diff
@@ -3,8 +3,12 @@ import React from 'react';
 export default function FormTitle({ title, subTitle }) {
   return (
     <div className="schemaform-title">
-      <h1>{title}</h1>
-      {subTitle && <div className="schemaform-subtitle">{subTitle}</div>}
+      <h1 data-testid="form-title">{title}</h1>
```
Comment (tlei123) [inline src/platform/forms-system/src/js/components/FormTitle.jsx:6]: Just adding data-testid HTML-attributes.

File: src/platform/forms-system/src/js/components/FormTitle.jsx [hunk 2]
```diff
@@ -3,8 +3,12 @@ import React from 'react';
...
+      {subTitle && (
+        <div className="schemaform-subtitle" data-testid="form-subtitle">
```
Comment (tlei123) [inline src/platform/forms-system/src/js/components/FormTitle.jsx:8]: Just adding data-testid HTML-attribute.

File: yarn.lock [hunk 3]
```diff
@@ -12277,7 +12277,7 @@ joycon@^2.2.5:
   resolved "https://registry.yarnpkg.com/joycon/-/joycon-2.2.5.tgz#8d4cf4cbb2544d7b7583c216fcdfec19f6be1615"
   integrity sha512-YqvUxoOcVPnCp0VU1/56f+iKSdvIRJYPznH22BdXV3xMk75SFXhWeJkZ8C9XxUWt1b5x2X1SxuFygW1U0FmkEQ==
 
-jpeg-js@0.2.0, jpeg-js@^0.4.0:
+jpeg-js@0.2.0, jpeg-js@^0.4.3:
```
Comment (tlei123) [inline yarn.lock:12280]: This change was automatic from running yarn install for updated package.json.

File: src/applications/hca/config/chapters/veteranInformation/americanIndian.js [hunk 4]
```diff
@@ -17,8 +16,11 @@ const Description = props => {
         need to pay a copay for care or services.
       </p>
 
-      <div className="vads-u-margin-top--3 vads-u-margin-bottom--7">
-        <AdditionalInfo triggerText="What it means to be recognized as an American Indian or Alaska Native">
+      <div
+        className="vads-u-margin-top--3 vads-u-margin-bottom--7"
+        data-testid="aiq-addl-info"
```
Comment (tlei123) [inline src/applications/hca/config/chapters/veteranInformation/americanIndian.js:21]: Adding data-testid HTML-attribute.

File: src/applications/hca/config/chapters/veteranInformation/americanIndian.js [hunk 5]
```diff
@@ -17,8 +16,11 @@ const Description = props => {
...
+      >
+        <va-additional-info trigger="What it means to be recognized as an American Indian or Alaska Native">
```
Comment (tlei123) [inline src/applications/hca/config/chapters/veteranInformation/americanIndian.js:23]: Subsituting newer component.

File: src/applications/hca/config/chapters/veteranInformation/americanIndian.js [hunk 6]
```diff
@@ -2,7 +2,6 @@ import fullSchemaHca from 'vets-json-schema/dist/10-10EZ-schema.json';
 
 import React from 'react';
 
-import AdditionalInfo from '@department-of-veterans-affairs/component-library/AdditionalInfo';
```
Comment (tlei123) [inline src/applications/hca/config/chapters/veteranInformation/americanIndian.js:5]: Substituting newer <va-additional-info> component.

File: src/applications/hca/tests/e2e/hca-aiq.cypress.spec.js [hunk 7]
```diff
@@ -47,9 +47,7 @@ describe('HCA-AIQ', () => {
   it('works with AIQ (Yes selected) - C12901', () => {
     cy.visit(manifest.rootUrl);
     cy.wait(['@mockUser', '@mockFeatures', '@mockEnrollmentStatus']);
-    cy.findAllByText(/apply.+health care/i, { selector: 'h1' })
-      .first()
-      .should('exist');
+    cy.findByTestId('form-title').contains(/apply.+health care/i);
```
Comment (k80bowman) [inline src/applications/hca/tests/e2e/hca-aiq.cypress.spec.js:50]: Is there a way to find these elements without adding a test ID? This isn't a pattern we use anywhere else in our tests and it's generally better to not add things to our code just for testing purposes.

Comment (tlei123) [reply src/applications/hca/tests/e2e/hca-aiq.cypress.spec.js:50]: It's a E2E best practice as [documented by Platform QA Standards team](https://depo-platform-documentation.scrollhelp.site/developer-docs/Cypress-Best-Practices-on-VA.gov-Platform.1870430211.html#CypressBestPracticesonVA.govPlatform-data-testidAttribute).  Cypress.io also has a [more general best-practice](https://docs.cypress.io/guides/references/best-practices#Selecting-Elements) incl. not selecting elements by text-content.  data-testid's just about the only way to select elements in a non-brittle way.  Hope you'll agree!  :)

Comment (k80bowman) [reply src/applications/hca/tests/e2e/hca-aiq.cypress.spec.js:50]: I apologize, I did not realize it was documented as a best practice. Thank you for pointing me to the documentation.

**Review by k80bowman — APPROVED** [review]

File: src/applications/hca/tests/e2e/hca-aiq.cypress.spec.js (hunk 7 above)
Comment (tlei123) [reply src/applications/hca/tests/e2e/hca-aiq.cypress.spec.js:50]: Thx much, Katie!  Resolving this convo.

File: yarn.lock (hunk 3 above)
Comment (theodur) [reply yarn.lock]: This was merged into `master` earlier today. I went ahead and merged `master` into your branch

Comment (tlei123) [reply yarn.lock]: Coolio!  Thx, Theo!  :)

**Review by micahchiang — APPROVED** [review]
lgtm, thanks @tlei123

**Review by lzapson — APPROVED** [review]
LGTM

**Review by Mottie — APPROVED** [review]

**Pull request merged**